wallet/src/de/schildbach/wallet/ui/dashpay/PlatformRepo.kt

```diff
@@ -142,9 +150,66 @@ class PlatformRepo(val walletApplication: WalletApplication) {
         }
     }
+
+    suspend fun initBlockchainIdentityData(username: String): BlockchainIdentityData {
+        return blockchainIdentityDataDaoAsync.load()
+                ?: BlockchainIdentityData(BlockchainIdentityData.State.UPGRADING_WALLET, false, username)
+    }
+
+    fun initBlockchainIdentity(blockchainIdentityData: BlockchainIdentityData, wallet: Wallet): BlockchainIdentity {
+        if (blockchainIdentityData.creditFundingTxId != null) {
+            val creditFundingTx = wallet.getTransaction(blockchainIdentityData.creditFundingTxId)
+            if (creditFundingTx != null) {
+                val creditFundingTransaction = wallet.getCreditFundingTransaction(creditFundingTx)
+                return BlockchainIdentity(Identity.IdentityType.USER, creditFundingTransaction, wallet).apply {
+                    currentUsername = blockchainIdentityData.username
+                    // should we load `preorderSalt` somehow?
+                    registrationStatus = blockchainIdentityData.registrationStatus!!
+                    // should we load `usernameStatus` somehow?
+                    // should we load `domain` somehow?
+                    creditBalance = blockchainIdentityData.creditBalance ?: Coin.ZERO
+                    activeKeyCount = blockchainIdentityData.activeKeyCount ?: 0
+                    totalKeyCount = blockchainIdentityData.totalKeyCount ?: 0
+                    keysCreated = blockchainIdentityData.keysCreated ?: 0
+                    currentMainKeyIndex = blockchainIdentityData.currentMainKeyIndex ?: 0
+                    currentMainKeyType = blockchainIdentityData.currentMainKeyType
+                            ?: IdentityPublicKey.TYPES.ECDSA_SECP256K1
+                }
+            }
+        }
+        return BlockchainIdentity(Identity.IdentityType.USER, 0, wallet)
+    }
+
+    suspend fun updateBlockchainIdentityData(blockchainIdentityData: BlockchainIdentityData, blockchainIdentity: BlockchainIdentity) {
+        blockchainIdentityData.apply {
+            if (blockchainIdentity.creditFundingTransaction != null) {
+                creditFundingTxId = blockchainIdentity.creditFundingTransaction!!.txId
+            }
+            registrationStatus = blockchainIdentity.registrationStatus
+            if (blockchainIdentity.currentUsername != null &&
+                    blockchainIdentity.registrationStatus == BlockchainIdentity.RegistrationStatus.REGISTERED) {
+                domain = Names.DEFAULT_PARENT_DOMAIN
+                username = blockchainIdentity.currentUsername
+                preorderSalt = blockchainIdentity.saltForUsername(blockchainIdentity.currentUsername!!, false)
+                usernameStatus = blockchainIdentity.statusOfUsername(blockchainIdentity.currentUsername!!)
+
+                // should we load `preorderSalt` somehow?
+                registrationStatus = blockchainIdentity.registrationStatus
+                // should we load `usernameStatus` somehow?
```

Comment:
We can use this code that was removed from a previous file:  https://github.com/dashevo/dash-wallet/pull/419/files#diff-52bbf8452a1888bfec68c85225efb675L198
```
blockchainIdentityData.preorderSalt = blockchainIdentity.saltForUsername(blockchainIdentity.currentUsername!!, false)
blockchainIdentityData.usernameStatus = blockchainIdentity.statusOfUsername(blockchainIdentity.currentUsername!!)
```